hdl21/external_module.py

```diff
@@ -39,7 +38,7 @@ class ExternalModule:
     paramtype: Type = HasNoParams  # Parameter-type `paramclass`
     desc: Optional[str] = None  # Description
     domain: Optional[str] = None  # Domain name, for references upon export
-    prefix: Optional[str] = None # Prefix that overrides default netlist naming
+    devicetype: Optional[str] = None # Device type, takes its cue from primitives.py
```

Comment:
I don't wanna add this.

Reply:
Yeah, that's fair, this is definitely not the HDL21 way of doing things. So I guess we could do the following:

1) Define a new device enum in primitives that basically just replicates VLSIR's SpicePrefix
2) Pass this enum to the ExternalModule's "property"
3) Make `to_proto` sift through properties looking for instances of the device enum
4) Tack on the enum's value onto the pkg protobuf as a param and let patched VLSIR work it's magic.

I'm just happy that I got those resistors to work! What do you think?

Reply:
RE: Latest Commit

- Added props to ExternalModule class to reflect Module class
- Persisted Sim simulator context into Tb.props
- Modified "to_proto" to look for SpicePrefix objects in Inst.module.props matched with simulator context in tops.module.props
- If no simulator is specified by the instance, default to SpicePrefix.SUBCKT
- If a simulator is specified but set to "NotImplemented" in the ExternalModule, throws a NotImplemented error. If simulator isn't described by ExternalModule, error will be thrown when program looks for said simulator key.
- This is then handled as before by passing the *value* of the chosen SpicePrefix to param inside the protobuf which is then parsed by vlsirtools into the correct spice prefix needed to run a component.